src/main/java/innohack/gem/service/extract/ExcelExtractor.java

```diff
@@ -1,13 +1,99 @@
 package innohack.gem.service.extract;
 
+import com.beust.jcommander.internal.Lists;
 import innohack.gem.entity.GEMFile;
 import innohack.gem.entity.extractor.ExtractConfig;
 import innohack.gem.entity.extractor.ExtractedRecords;
+import innohack.gem.entity.extractor.TimestampColumn;
+import innohack.gem.entity.feature.AbstractFeature;
+import innohack.gem.entity.feature.ExcelFeature;
+import innohack.gem.entity.rule.ParamType;
+import innohack.gem.entity.rule.Parameter;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 public class ExcelExtractor extends AbstractExtractor {
 
+  public ExcelExtractor() {
+    this(null, null);
+  }
+
+  public ExcelExtractor(String sheetName, String columnNames) {
+    Parameter param1 = new Parameter("Sheet Name", "", ParamType.STRING_LIST, sheetName);
+    Parameter param2 = new Parameter("Column Names", "", ParamType.STRING_LIST, columnNames);
+
+    setParams(Lists.newArrayList(param1, param2));
+  }
+
   @Override
   public ExtractedRecords extract(GEMFile f, ExtractConfig extractConfig) throws Exception {
-    return null;
+    f.extract();
+
+    for (AbstractFeature feature : f.getData()) {
+
+      if (feature instanceof ExcelFeature) {
+        ExtractedRecords results = new ExtractedRecords();
+
```

Comment:
This should be sheetname

Reply:
meaning in extractedrecords, there is another var called sheetname?

Reply:
recommit the codes.. see if this is the one u looking for?